identity/src/test/java/com/android/identity/MobileSecurityObjectParserTest.java

```diff
@@ -0,0 +1,98 @@
+package com.android.identity;
+
+import androidx.test.filters.SmallTest;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.math.BigInteger;
+import java.security.PublicKey;
+import java.security.cert.CertificateEncodingException;
+import java.util.Map;
+import java.util.Set;
+
+import co.nstant.in.cbor.model.DataItem;
+
+public class MobileSecurityObjectParserTest {
+
+    @Test
+    @SmallTest
+    public void testMSOParserWithVectors() throws CertificateEncodingException {
+        DataItem deviceResponse = Util.cborDecode(Util.fromHex(
+                TestVectors.ISO_18013_5_ANNEX_D_DEVICE_RESPONSE));
+        DataItem documentDataItem = Util.cborMapExtractArray(deviceResponse,
+                "documents").get(0);
+
+        DataItem issuerSigned = Util.cborMapExtractMap(documentDataItem, "issuerSigned");
+        DataItem issuerAuthDataItem = Util.cborMapExtract(issuerSigned, "issuerAuth");
+
+        DataItem mobileSecurityObjectBytes = Util.cborDecode(
+                Util.coseSign1GetData(issuerAuthDataItem));
+        DataItem mobileSecurityObject = Util.cborExtractTaggedAndEncodedCbor(
+                mobileSecurityObjectBytes);
+        byte[] encodedMobileSecurityObject = Util.cborEncode(mobileSecurityObject);
+
+        MobileSecurityObjectParser.MobileSecurityObject mso = new MobileSecurityObjectParser()
```

Comment:
Maybe add a comment saying that the response above and all the following constants are from _ISO 18013-5 D.4.1.2 mdoc response_ and the goal here is to check that the parser returns the expected values... I think that might help the reader figure out how it works!